### PR TITLE
samples: usb: Make each sample have its own USB product string

### PR DESCRIPTION
**Pull request by IvanSanchez**
Each USB sample, when compiled, will provide a different
USB product string. A "nice to have" feature when trying
out the USB samples (they will show up differently in
`lsusb` or `dmesg`), and also illustrates how to set up
custom USB constants in the config files.

Requires #6589.

Signed-off-by: Iván Sánchez Ortega <ivan@sanchezortega.es>

### CODE REVIEW

File: samples/subsys/usb/hid/prj.conf [hunk 1]
```diff
@@ -1,6 +1,7 @@
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
 CONFIG_USB_DEVICE_HID=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr Human Interface Device sample"
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/hid/prj.conf]: I would prefer something shorter, s/Human Interface Device/HID

File: samples/subsys/usb/mass/prj.conf [hunk 2]
```diff
@@ -7,6 +7,7 @@ CONFIG_GPIO=y
 #USB related configs
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr USB mass storage sample"
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/mass/prj.conf]: s/mass storage/MSC

File: samples/subsys/usb/dfu/prj.conf [hunk 3]
```diff
@@ -3,6 +3,7 @@ CONFIG_GPIO=y
 CONFIG_ARC_INIT=n
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr USB DFU sample"
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/dfu/prj.conf]: s/USB DFU/DFU

File: samples/subsys/usb/cdc_acm/prj.conf [hunk 4]
```diff
@@ -4,6 +4,7 @@ CONFIG_GPIO=y
 CONFIG_ARC_INIT=n
 CONFIG_USB=y
 CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr USB CDC ACM sample"
```
Comment (jfischer-phytec-iot) [inline samples/subsys/usb/cdc_acm/prj.conf]: s/USB CDC/CDC

**Review by jfischer-no — DISMISSED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**